drivers/util.py

```diff
@@ -1098,24 +1097,30 @@ def dom0_disks():
     return disks
 
 
-def set_scheduler_sysfs_node(node, str):
-    """Set the scheduler for a sysfs node (e.g. '/sys/block/sda')"""
+def set_scheduler_sysfs_node(node, scheds):
+    """
+    Set the scheduler for a sysfs node (e.g. '/sys/block/sda')
+    according to prioritized list schedulers
+    Try to set the first item, then fall back to the next on failure
```

Comment:
Instead of trial and error the code could read the list of available schedulers, and take the intersection with the requested ones and do a single call that will be known to succeed (or raise an error message printing the list of available schedulers vs requested scheduler for easier debugging)

Reply:
@edwintorok  I have re-implemented  basing on the advice,  But I found following problem.

**The sysfs node does not show all available schedulers**,  (some supported schedulers are pluggable, and need to be loaded)
for example, 
```
cat /sys/block/sda/queue/scheduler 
[mq-deadline] none

 modprobe bfq
 cat /sys/block/sda/queue/scheduler
[mq-deadline] bfq none

=====================
cat /sys/block/sda/queue/scheduler
[mq-deadline] none

echo bfq > /sys/block/sda/queue/scheduler
cat /sys/block/sda/queue/scheduler
mq-deadline [bfq] none
```
Instead of doing other search/on-demand detect, I prefer to set as detect directly.